samples/extensions/open_gl_interop/open_gl_interop.h

```diff
@@ -0,0 +1,114 @@
+/* Copyright (c) 2019-2020, Bradley Austin Davis
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "api_vulkan_sample.h"
+#include "rendering/render_pipeline.h"
+#include "scene_graph/components/camera.h"
+
+class OpenGLWindow;
+
+#if defined(WIN32)
+using Handle = HANDLE;
+#else
+using Handle                          = int;
+constexpr Handle INVALID_HANDLE_VALUE = static_cast<Handle>(-1);
+#endif
+
+// Vertex layout for this example
+struct VertexStructure
+{
+	float pos[3];
+	float uv[2];
+	float normal[3];
+};
+
+class OpenGLInterop : public ApiVulkanSample
+{
+	friend class OpenGLWindow;
```

Comment:
A lot of the functionality of this friend class can be moved into `OpenGLInterop` making it less useful. I think at some point custom windows will be needed for different platforms to initialize opengl but currently the friend class confuses things. My previous comments mention moving functionality back into `OpenGLInterop`

Reply:
Moving the friend class functionality into the main sample class pretty much requires that I move all the GLAD / EGL headers from `open_gl_interop.cpp` into `open_gl_interop.h`.  Alternatively I could move the functions as you suggest and have `open_gl_interop.h` forward declare a `struct OpenGLState` type and declare a pointer to it.   But even then, moving a function like `static void debugMessageCallback(GLenum source, GLenum type, GLuint id, GLenum severity, GLsizei length, const GLchar *message, const void *userParam)` into the header implies that `GLenum`, `GLsizei`, etc are all exposed.  

I don't really anticipate other examples needing to use OpenGL so I wanted to encapsulate it's use to this one PR as much as possible.